### ISSUE TITLE
The example of "libvox" doesn't work for me.

### ISSUE THREAD

**Issue by rempas** [MrSmith33/vox#45]
**Expected behavior**
I was expecting the program to work (at least for the D version).

**Describe the bug**
The example doesn't work.

**To Reproduce**
I'm trying to compile the libvox [example](https://github.com/MrSmith33/vox/blob/master/example/vox_exe.d) but I have no luck. I'm trying to do it both using D (and specifically LDC) and by using C. In the C version, I also had to copy the decorations from the "libvox" module and change some stuff to make it work with a C compiler. For that reason, there may be mistakes that I made here. Having "libvox" work with a C program is very important as pretty much everyone that's gonna make a language frontend will want to bootstrap at some point so it makes sense for me to test it and report back.

In the C version, the program compiles normally but when I try to run it, I get a segmentation fault. In the D version, I'm getting linker errors. More specifically:

```
/usr/bin/ld: test_libvox_d.o: in function `main':
test_libvox.d:(.text.main+0x4c): undefined reference to `_D6libvox__T5SliceTxaZQk6__ctorMFNaNbNcNiAxaZSQBs__TQBoTxaZQBv'
/usr/bin/ld: test_libvox.d:(.text.main+0x98): undefined reference to `_D6libvox__T5SliceTxaZQk6__ctorMFNaNbNcNiAxaZSQBs__TQBoTxaZQBv'
/usr/bin/ld: test_libvox.d:(.text.main+0xcf): undefined reference to `_D6libvox__T5SliceTxaZQk6__ctorMFNaNbNcNiAxaZSQBs__TQBoTxaZQBv'
collect2: error: ld returned 1 exit status
Error: /usr/bin/cc failed with status: 1
```

I have uploaded the files in a [repo](https://codeberg.org/rempas/libvox_error) so you can easily test it out yourself and examine the code and see if I made any mistakes (if you prefer to do so). You can see more info in the 'README.md' in the repo. However, of course ask me if you need anything. I'll try to see and answer as fast as I can :)

**Additional context**
 - OS: Linux
 - [c73cd86](https://github.com/MrSmith33/vox/commit/c73cd86c73e31ba17896b8bb4eb44695833df582)

**Comment by MrSmith33** [comment]
The compiler says undefined reference to Slice constructor. That is because you didn't include libvox.d into the compilation.
Another thing to note is that prebuilt libvox.so statically links to druntime, which may cause a problem when loaded into non-better-c D program, but check this anyway.
The intended use of libvox from regular D is either using source code or though dub package. I should document all this in the future.

**Comment by rempas** [comment]
Thank you! It now compiles! However, just like the C version of the program, I now get the same output as the C program. Which is the "segmentation fault". I copy-pasted the files from the "example" [directory](https://github.com/MrSmith33/vox/tree/master/example) so I don't see any way I could mess up (but from my history, I would not be surprised 😅).

**Comment by MrSmith33** [comment]
git clone gives me `fatal: unable to access 'https://codeberg.org/rempas/libvox_error.git/': server certificate verification failed. CAfile: none CRLfile: none`
Can you run the program under the gdb and print a stacktrace?
I suspect that the problem may be with libc that it was linked with during CI build. It was built on Ubuntu 20.4. What's your OS?

**Comment by MrSmith33** [comment]
You can try building the .so file yourself with
```
vox/source> ldc2 -m64 -shared -O3 -mcpu=x86-64-v3 --d-debug -g -boundscheck=off -enable-inlining -flto=full -linkonce-templates --defaultlib=phobos2-ldc-lto,druntime-ldc-lto -fvisibility=hidden -link-defaultlib-shared=false -i c_api.d -of=../bin/libvox.so
```

**Comment by MrSmith33** [comment]
In case this issue is the same as #46, and I guessed that the problem is `-mcpu` flag, you can try compiling without `-mcpu=x86-64-v3`

**Comment by MrSmith33** [comment]
And I should probably build a version with lower CPU requirement too

**Comment by MrSmith33** [comment]
What's you compiler version?

**Comment by rempas** [comment]
Good morning! I have tested out on my work's laptop without the `mcpu=x86-64-v3` flag and it works without problems! I suppose this also fixes #46 (I haven't tested it out tho) so I will close that too! Once again, thank you for your help!
